libs/eavmlib/src/http_server.erl

```diff
@@ -77,7 +77,10 @@ reply(StatusCode, ReplyBody, Conn) ->
         StatusCode, ReplyBody, [<<"Content-Type: text/html\r\nConnection: close\r\n">>], Conn
     ),
     Socket = proplists:get_value(socket, NewConn),
-    gen_tcp:close(Socket),
+    spawn_link(fun() ->
+        timer:sleep(5000),
```

Comment:
This sounds quite odd, we are hanging for 5 seconds before closing the socket. 
Do we have any issue about the root cause we are trying to fix?

Reply:
> We shouldn't close the socket the way we do it, even with `Connection: close` header. This is against the protocol.
> 
> We could implement something with a loop reading for client data following what nginx does.
> 
> https://nginx.org/en/docs/http/ngx_http_core_module.html#lingering_close

Exactly, I thought it was supposed to be the clients job to close the connection… and most servers will wait ip to five minutes of no activity from the client before closing a socket. We obviously cannot afford to wait this long if the server is an MCU with a limited number of sockets, or it will be very quickly become unavailable for any new connections. Waiting 5 to 15 seconds seems reasonable, but maybe that should be configurable when the server is started.

Reply:
Further reading I see it is the servers job to close the socket to let the client know there is no further data, but if it is closed too quickly the client will get a TCP RST, but we should wait a short amount of time during which we make sure the client isn’t in the middle of making another request.

Reply:
What I have submitted is actually not right at all. Now we would be making the client open a new socket to load a jpg for example, and on an esp32 with only 8-10 sockets available at a time all sockets could quickly be exhausted and stuck in that 5000 ms waiting period.

Reply:
> Do sockets still count in this 8-10 quota you mention once the client closes them?

Closed sockets should not count. The maximum number of open sockets defaults to 10, but can be increased as a KConfig option (at the expense of more ram consumed by lwip):
https://github.com/atomvm/AtomVM/pull/1496#discussion_r1932649364